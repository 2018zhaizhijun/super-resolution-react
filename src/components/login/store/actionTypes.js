export const CHANGE_IS_VISIBLE = 'login/CHANGE_IS_VISIBLE';
//export const CHANGE_REGISTER_STATE = 'login/CHANGE_REGISTER_STATE';
export const CHANGE_LOGIN_STATE = 'login/CHANGE_USER_LOGIN_STATE';
export const CHANGE_AVATAR = 'login/CHANGE_AVATAR';
export const CHANGE_PROFILE = 'login/CHANGE_PROFILE_INFO';
export const CHANGE_TOKEN = 'login/CHANGE_PROFILE_TOKEN';
