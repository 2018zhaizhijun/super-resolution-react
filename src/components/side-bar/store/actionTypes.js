export const CHANGE_IS_VISIBLE = 'sidebar/CHANGE_IS_VISIBLE';
