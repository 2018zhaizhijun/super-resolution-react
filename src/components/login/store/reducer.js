import { Map } from 'immutable';
import * as actionTypes from './actionTypes';

const defaultState = Map({
    isVisible: false,
    //registering: false,
    isLogin: false,
    profile: {
        avatarUrl: null,
        usrname: null,
        telephone: null,
        sex: 0,
    },
    token: null,
});

function reducer(state=defaultState, action) {
    switch (action.type) {
        case actionTypes.CHANGE_IS_VISIBLE:
            return state.set('isVisible', action.isVisible);
        // case actionTypes.CHANGE_REGISTER_STATE:
        //     return state.set('registering', action.registering);
        case actionTypes.CHANGE_LOGIN_STATE:
            return state.set('isLogin', action.isLogin);
        // case actionTypes.CHANGE_AVATAR:
        //     return state.set('avatarUrl', action.avatarUrl);
        case actionTypes.CHANGE_PROFILE:
            return state.set('profile', {...state.get('profile'), ...action.profile});
        case actionTypes.CHANGE_TOKEN:
            return state.set('token', action.token);
        default:
            return state;
    }
}

export default reducer;