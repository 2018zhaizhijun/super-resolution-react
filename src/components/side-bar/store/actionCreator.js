import * as actionTypes from './actionTypes';
import { createAction } from '@reduxjs/toolkit';

export const changeIsVisible = createAction(actionTypes.CHANGE_IS_VISIBLE);
// export const changeIsVisible = (visibleState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     isVisible: visibleState
// });
