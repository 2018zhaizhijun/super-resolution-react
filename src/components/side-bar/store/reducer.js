import { Map } from 'immutable';
import * as actionTypes from './actionTypes';

const defaultState = Map({
    isVisible: false,
});

function reducer(state=defaultState, action) {
    switch (action.type) {
        case actionTypes.CHANGE_IS_VISIBLE:
            return state.set('isVisible', action.isVisible);
        default:
            return state;
    }
}

export default reducer;