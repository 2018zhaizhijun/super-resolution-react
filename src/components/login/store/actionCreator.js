import { phoneLogin, getAvatar, getProfile, logout, sendAvatar } from '@service/login';
import * as actionTypes from './actionTypes';
import { message } from 'antd';

import { sendUsrinfo } from '@service/login';

export const changeIsVisible = (visibleState) => ({
    type: actionTypes.CHANGE_IS_VISIBLE,
    isVisible: visibleState
});

// export const changeRegisterState = (registerState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     registering: registerState
// });

export const changeLoginState = (loginState) => ({
    type: actionTypes.CHANGE_LOGIN_STATE,
    isLogin: loginState
});

// export const changeAvatar = (avatarUrl) => ({
//     type: actionTypes.CHANGE_AVATAR,
//     avatarUrl
// });

export const changeProfile = (profile) => ({
    type: actionTypes.CHANGE_PROFILE,
    profile
});

export const changeToken = (token) => ({
    type: actionTypes.CHANGE_PROFILE,
    token
});

export const performLogin = (telephone, password) => {
    return (dispatch) => {
        phoneLogin(telephone, password).then((response) => {
            console.log(response);
            if (response.statusCode != 200) {
                message.error('账号或密码错误');
            }
            else {
                const token = response.data.token;
                dispatch(changeLoginState(true));
                dispatch(changeToken(token));

                // sendUsrinfo('', 'zz', 1).then((r)=>{
                //     console.log(r);
                //     if (r.statusCode == 200) {
                getProfile().then((res) => {
                    console.log(res);
                    if (res.statusCode != 200) {
                        message.error('用户信息获取失败');
                    }
                    else {
                        let data = res.data;
                        dispatch(changeProfile({
                            avatarUrl: data.avatar,
                            usrname: data.trueName,
                            telephone,
                            sex: data.sex,
                        }));
                    }
                });

                // getAvatar().then((res) => {
                //     console.log(res);
                //     dispatch(changeAvatar(res))
                //     // if (res.statusCode != 200) {
                //     //     message.error('用户头像获取失败');
                //     // }
                //     // else {
                //     //     dispatch(changeAvatar(res.data.avatar));
                //     // }
                // });

                dispatch(changeIsVisible(false));
                //     }
                    
                // });    
            }
        })
    };
};

export const performLogout = () => {
    return (dispatch) => {
        logout().then( (res) => {
            if (res.statusCode == 200) {
                dispatch(changeLoginState(false));
            }
            else {
                message.error('退出登录失败');
            }
        });
    };
};

// export const updateAvatar = (avatarFile) => {
//     return (dispatch) => {
//         sendAvatar(avatarFile).then( (res) => {
//             if (res.statusCode == 200) {
//                 dispatch(changeAvatar(res.data.avatarUrl));
//             }
//             else {
//                 message.error('头像上传失败');
//             }
//         });
//     };
// };
