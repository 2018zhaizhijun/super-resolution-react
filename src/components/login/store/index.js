import reducer from './reducer';
import { changeIsVisible, changeProfile, performLogin, performLogout } from './actionCreator';

export {
    reducer,
    changeIsVisible,
    changeProfile,
    performLogin,
    performLogout,
    //updateAvatar,
}