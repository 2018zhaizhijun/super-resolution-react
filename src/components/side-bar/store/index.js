import reducer from './reducer';
import { changeIsVisible as changeSidebarVisible } from './actionCreator';

export {
    reducer,
    changeSidebarVisible,
}