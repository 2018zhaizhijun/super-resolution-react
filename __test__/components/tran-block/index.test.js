import React from 'react';
import TransBlock from '@components/tran-block';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '@store';

describe('transform block render', () => {

  it('transform block should match snapshot', () => {
    const transWrapper = renderer.create(
        <Provider store={store}>
            <TransBlock />
        </Provider>
    ).toJSON();
    expect(transWrapper).toMatchSnapshot();
  });

})