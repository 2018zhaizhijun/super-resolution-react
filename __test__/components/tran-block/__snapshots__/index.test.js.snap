// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform block render transform block should match snapshot 1`] = `
<div
  className="sc-bdvvtL coAMqL"
>
  <div
    className="draggerArea"
    id="draggerWrapper"
    onClickCapture={[Function]}
  >
    <span>
      <div
        className="ant-upload ant-upload-drag"
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "height": undefined,
          }
        }
      >
        <span
          className="ant-upload ant-upload-btn"
          onClick={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <input
            accept="image/*,video/*"
            multiple={true}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
          <div
            className="ant-upload-drag-container"
          >
            <p
              className="ant-upload-drag-icon"
            >
              <span
                aria-label="upload"
                className="anticon anticon-upload"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="upload"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </p>
            <p
              className="ant-upload-text"
            >
              点击或拖拽上传图片或视频
            </p>
            <p
              className="ant-upload-hint"
            >
              文件格式：.jpg .png .mp4 .webm .ogg
              <br />
              文件大小限制：2MB
            </p>
          </div>
        </span>
      </div>
    </span>
  </div>
  <div
    className="draggerArea resultArea"
  >
    <span>
      <div
        className="ant-upload ant-upload-drag displayDragger"
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "height": undefined,
          }
        }
      >
        <span
          className="ant-upload ant-upload-btn"
          onClick={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <input
            accept=""
            multiple={false}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
          <div
            className="ant-upload-drag-container"
          />
        </span>
      </div>
      <div
        className="ant-upload-list ant-upload-list-picture"
      />
    </span>
  </div>
</div>
`;
